File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestPlanner.cs
using Microsoft.Bot.Builder;
using Microsoft.Teams.AI.AI;
using Microsoft.Teams.AI.AI.Planners;
using Microsoft.Teams.AI.State;
using System.Reflection;

namespace Microsoft.Teams.AI.Tests.TestUtils
{
    public class TestPlanner : IPlanner<TurnState>
    {
        public IList<string> Record { get; } = new List<string>();

        public Plan BeginPlan { get; set; } = new Plan
        {
            Commands = new List<IPredictedCommand>
            {
                new PredictedDoCommand("Test-DO"),
                new PredictedSayCommand("Test-SAY")
            }
        };

        public Plan ContinuePlan { get; set; } = new Plan
        {
            Commands = new List<IPredictedCommand>
            {
                new PredictedDoCommand("Test-DO"),
                new PredictedSayCommand("Test-SAY")
            }
        };

        public Task<Plan> BeginTaskAsync(ITurnContext turnContext, TurnState turnState, AI<TurnState> ai, CancellationToken cancellationToken)
        {
            Record.Add(MethodBase.GetCurrentMethod()!.Name);
            return Task.FromResult(BeginPlan);
        }

        public Task<Plan> ContinueTaskAsync(ITurnContext turnContext, TurnState turnState, AI<TurnState> ai, CancellationToken cancellationToken)
        {
            Record.Add(MethodBase.GetCurrentMethod()!.Name);
            return Task.FromResult(ContinuePlan);
        }
    }
}

File: js/packages/teams-ai/src/planners/TestPlanner.ts
import { TurnContext } from 'botbuilder';
import { Plan, Planner, PredictedSayCommand } from './Planner';
import { TurnState } from '../TurnState';
import { AI } from '../AI';

/**
 * A planner used for testing.
 */
export class TestPlanner implements Planner<TurnState> {
    public constructor(beginPlan?: Plan, continuePlan?: Plan) {
        this.beginPlan = beginPlan || {
            type: 'plan',
            commands: [{ type: 'SAY', response: 'Hello World' } as PredictedSayCommand]
        };
        this.continuePlan = continuePlan || {
            type: 'plan',
            commands: []
        };
    }

    public readonly beginPlan: Plan;
    public readonly continuePlan: Plan;

    public beginTask(
        context: TurnContext,
        state: TurnState,
        ai: AI<TurnState>
    ): Promise<Plan> {
        return Promise.resolve(this.beginPlan);
    }

    public continueTask(
        context: TurnContext,
        state: TurnState,
        ai: AI<TurnState>
    ): Promise<Plan> {
        return Promise.resolve(this.continuePlan);
    }
}
